app/controllers/provider_interface/organisation_permissions_setup_controller.rb

```diff
@@ -54,10 +54,26 @@ def update
 
     def check
       wizard = OrganisationPermissionsSetupWizard.new(organisation_permissions_wizard_store, current_step: :check)
+      wizard.save_state!
       @relationships = wizard.relationships
       @previous_page_path = previous_page_path(wizard)
     end
 
+    def commit
+      wizard = OrganisationPermissionsSetupWizard.new(organisation_permissions_wizard_store)
+      if SetupProviderRelationshipPermissions.call(wizard.relationships)
+        wizard.clear_state!
+        redirect_to success_provider_interface_organisation_permissions_setup_index_path
+      else
+        redirect_to(
```

Comment:
do we want to clear the state here as well, since if we get a validation error at this stage its fair to assume the state is corrupt?

Reply:
Yes, but we will need to get a better warning message from the design team, as it's currently expecting them to go back to the check page, which would need something in the state store

Reply:
should we redirect to the start page instead?

Reply:
It now redirects to the start page for the setup flow

Reply:
could we make sure to also call `wizard.clear_state!` as well here?